crates/primitives/src/header.rs

```diff
@@ -554,13 +554,14 @@ impl Decodable for Header {
 
 /// A [`Header`] that is sealed at a precalculated hash, use [`SealedHeader::unseal()`] if you want
 /// to modify header.
-#[add_arbitrary_tests(rlp)]
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[main_codec(no_arbitrary)]
+#[add_arbitrary_tests(rlp, compact)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct SealedHeader {
+    /// Locked Header hash.
+    pub hash: B256,
```

Comment:
why not continue using `BlockHash` alias for `B256`?

Reply:
It requires adding `BlockHash` to `Compact` known types macro, which I was avoiding. But I noticed that there's `TxHash` there already, so why not. fixed